File: sdk/swift/Sources/ZcashNameSDK/Pricing.swift
import Foundation

/// Compute the claim cost in zatoshis for a name of the given length.
///
/// - Parameters:
///   - tiers: The pricing tiers from `StatusResult.pricing.tiers` (already in zatoshis).
///            Index 0 = 1-char names, index 1 = 2-char, etc.
///            Names longer than the array clamp to the last entry.
///   - nameLength: Character length of the name (must be >= 1).
/// - Returns: Cost in zatoshis, or `nil` if tiers is empty.
public func claimCost(tiers: [Int], nameLength: Int) -> Int? {
    guard !tiers.isEmpty else { return nil }
    let idx = min(max(nameLength - 1, 0), tiers.count - 1)
    return tiers[idx]
}

File: sdk/python/zcashname/pricing.py
from __future__ import annotations


def claim_cost(tiers: list[int], name_length: int) -> int | None:
    """Compute the claim cost in zatoshis for a name of the given length.

    *tiers* comes from ``StatusResult.pricing.tiers`` (already in zatoshis).
    Index 0 corresponds to 1-char names, index 1 to 2-char names, etc.
    Names longer than the array clamp to the last entry.

    Returns ``None`` if *tiers* is empty.
    """
    if not tiers:
        return None
    idx = min(max(name_length - 1, 0), len(tiers) - 1)
    return tiers[idx]
